GiniVision/Classes/ToolTipView.swift

```diff
@@ -0,0 +1,303 @@
+//
+//  ToolTipView.swift
+//  GiniVision
+//
+//  Created by Enrique del Pozo Gómez on 9/21/17.
+//
+
+import Foundation
+import UIKit
+
+final class ToolTipView: UIView {
+    
+    enum ToolTipPosition {
+        case above
+        case below
+        case left
+        case right
+    }
+    
+    fileprivate var arrowWidth:CGFloat = 20
+    fileprivate var arrowHeight:CGFloat = 20
+    fileprivate var closeButtonWidth:CGFloat = 20
+    fileprivate var closeButtonHeight:CGFloat = 20
+    fileprivate var itemSeparation: CGFloat = 16
+    fileprivate var margin:(top:CGFloat, left:CGFloat, right: CGFloat, bottom: CGFloat) = (20, 20, 20, 20)
+    fileprivate var maxWidthOnIpad:CGFloat = 375
+    fileprivate var padding:(top:CGFloat, left:CGFloat, right: CGFloat, bottom: CGFloat) = (16, 16, 16, 16)
+    
+    fileprivate var textWidth:CGFloat {
+        guard let superview = superview else { return 0 }
+        if UIDevice.current.isIpad {
+            return maxWidthOnIpad - padding.left - padding.right - margin.left - margin.right - closeButtonWidth - itemSeparation
+        } else {
+            return superview.frame.width - padding.left - padding.right - margin.left - margin.right - closeButtonWidth - itemSeparation
+        }
```

Comment:
Could be simplified and clarified a bit:
```
let maxWidth = UIDevice.current.isIpad ? maxWidthOnIpad : superview.frame.width
let padding = padding.left + padding.right
let margin = margin.left + margin.right
return maxWidth - padding - margin - closeButtonWidth - itemSeparation
```

Reply:
I did a mix of both, since Xcode complains because `padding` property has the same as the other one. (I have preferred to not combining instead of using other different name like _paddingSides_ or similar)